client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx

```diff
@@ -0,0 +1,463 @@
+
+    // Update the priceByRowId
+    setPriceByRowId((prevState) => ({
+      ...prevState,
+      [row.id]: price.toFixed(2),
+    }));
+  };
+
+  const handlePriceChange = (event, row) => {
+    if (row.complementary) return; // If complementary, no changes allowed
+    const newPrice = event.target.value;
+    setPriceByRowId((prevState) => ({
+      ...prevState,
+      [row.id]: newPrice,
+    }));
+  };
+
+  const handlePriceBlur = (event, row) => {
+    let newPrice = parseFloat(event.target.value).toFixed(2);
+    if (isNaN(parseFloat(newPrice))) {
+      newPrice = '0.00';
+    }
+    setPriceByRowId((prevState) => ({
+      ...prevState,
+      [row.id]: newPrice,
+    }));
+  };
+
+  const handleComplementaryChange = (event, row) => {
+    const isChecked = event.target.checked;
+    // Update row with complementary flag
+    const updatedRows = eventData.map((r) => {
+      if (r.id === row.id) {
+        return {
+          ...row,
+          complementary: isChecked,
+          price: isChecked ? 0 : row.price,
+        };
+      }
+      return r;
+    });
+    setEventData(updatedRows);
+    // Also update the priceByRowId to reflect the change in the price
+    if (isChecked) {
+      setPriceByRowId((prevState) => ({
+        ...prevState,
+        [row.id]: '0.00', // Set to $0 if complementary
+      }));
+    }
+  };
+
+  const handlePurchase = () => {
+    // Redirect to checkout page
+  };
+
+  const getEventData = async (event) => {
+    try {
+      const token = await getAccessTokenSilently({
+        audience: process.env.REACT_APP_ROOT_URL,
+        scope: 'admin',
+      });
+      const response = await fetch(
+        process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${event}`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${token}`,
+          },
+        },
+      );
+      const jsonRes = await response.json();
+      const jsonData = jsonRes.data;
+      const eventData = jsonData
+        .map((item, index) => {
+          const row = item.row.slice(1, -1).split(',');
+          if (!row || !row[0]) {
+            // Check if the value in column 0 is present
+            setTicketsSold(false);
+            return null; // Exit early if no tickets sold
+          }
+          setTicketsSold(true);
+          return {
+            id: index,
+            firstname: row[1],
+            lastname: row[2],
+            num_tickets: row[12],
+            arrived: false,
+            vip: row[3] === 't',
+            donorbadge: row[4] === 't',
+            accommodations: row[5],
+          };
+        })
+        .filter(Boolean);
+
+      setEventData(eventData);
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+
+  return (
+    <div className='w-full h-screen overflow-x-hidden absolute '>
+      <div className='md:ml-[18rem] md:mt-40 sm:mt-[11rem] sm:ml-[5rem] sm:mr-[5rem] sm:mb-[11rem]'>
```

Comment:
Could you change the classes on this div to `md:ml-[18rem] md:mt-40 md:mb-[11rem] tab:mx-[5rem] mx-[1.5rem] my-[9rem]`. When I went through all of the admin pages a couple weeks back, each one of the pages got these styles so it looks a bit more pleasant on mobile

Reply:
updated.